docs/cephfs-snap-clone.md

```diff
@@ -0,0 +1,79 @@
+# Create CephFS snapshot and Clone Volume
+
+- [Create CephFS snapshot and Clone Volume](#create-cephfs-snapshot-and-clone-volume)
+  - [Prerequisite](#prerequisite)
+  - [Snapshot](#snapshot)
+    - [Create snapshotclass](#create-snapshotclass)
+    - [Create snapshot](#create-snapshot)
+  - [Restore snapshot to a new PVC](#restore-snapshot-to-a-new-pvc)
+  - [Clone PVC](#clone-pvc)
+
+## Prerequisite
+
+- For snapshot functionality to be supported for your Kubernetes cluster, the
+  Kubernetes version running in your cluster should be >= v1.17. We also need the
+  `snapshot controller` deployed in your Kubernetes cluster along with `csi-snapshotter`
+  sidecar container.
+  Refer [external-snapshotter](https://github.com/kubernetes-csi/external-snapshotter/)
+  for more information on these sidecar controllers. There should
+  be a `volumesnapshotclass` object present in the cluster
+  for snapshot request to be satisified.
+
+## Snapshot
+
+### Create snapshotclass
+
+```console
+$kubectl create -f snapshotclass.yaml
+```
+
+### Create snapshot
+
+The snapshot is created on/for an existing PVC. You should
+have a PVC in bound state before creating snapshot from it.
+please refer pvc creation [doc](https://github.com/ceph/ceph-csi/blob/master/docs/deploy-cephfs.md)
+for more information on how to create a PVC.
+
+```console
+$ kubectl create -f snapshot.yaml
+volumesnapshot.snapshot.storage.k8s.io/cephfs-pvc-snapshot created
+```
+
+- Get details about the snapshot
+
+```console
+$ kubectl get volumesnapshot
+NAME                  READYTOUSE   SOURCEPVC    SOURCESNAPSHOTCONTENT   RESTORESIZE   SNAPSHOTCLASS                SNAPSHOTCONTENT                                    CREATIONTIME   AGE
+cephfs-pvc-snapshot   true         csi-cephfs-pvc                          1Gi           csi-cephfsplugin-snapclass   snapcontent-34476204-a14a-4d59-bfbc-2bbba695652c   3s             6s
+```
+
+- Get details about the volumesnapshotcontent
+
+```console
+$ kubectl get volumesnapshotcontent
+NAME                                               READYTOUSE   RESTORESIZE   DELETIONPOLICY   DRIVER                          VOLUMESNAPSHOTCLASS          VOLUMESNAPSHOT        AGE
+snapcontent-34476204-a14a-4d59-bfbc-2bbba695652c   true         1073741824    Delete           rook-ceph.cephfs.csi.ceph.com   csi-cephfsplugin-snapclass   cephfs-pvc-snapshot   64s
+$
```

Comment:
@nixpanic  removed extra `$`. Regarding one instance of `$kubectl` it was kept that way because there is no output to that command . Without out md linter will complain :+1:

Reply:
in that case, we don't need to use $ see https://github.com/ceph/ceph-csi/tree/master/examples

Reply:
sure. Done.

Reply:
Sorry, I don't get it. `$` is still there, and there are still inconsistencies with `kubectl`, `$ kubectl` (this document) and `[$] kubectl` (from other docs).

Reply:
This was marked as resolved, but I do not think it is?